test/functional/feature_anchors.py

```diff
@@ -78,6 +81,60 @@ def run_test(self):
         self.log.info("When node starts, check if anchors.dat doesn't exist anymore")
         assert not os.path.exists(node_anchors_path)
 
+        self.log.info("Ensure addrv2 support")
+        # Use proxies to catch outbound connections to networks with 256-bit addresses
+        onion_conf = Socks5Configuration()
+        onion_conf.auth = True
+        onion_conf.unauth = True
+        onion_conf.addr = ('127.0.0.1', 10000)
+        onion_conf.keep_alive = True
```

Comment:
Why do we need to keep the `Socks5Server` alive to ensure success when performing `"Check for addrv2 addresses in anchors.dat"`?

I changed this to `False` and saw intermittent failures on Line 123

```
assert_equal(data[services_index], 0x00) # services == NONE
```

Reply:
Good question. The Socks5 server we use in the test framework by default disconnects immediately after the handshake and sometimes would not register as a connected peer:

https://cirrus-ci.com/task/6075793026056192?logs=ci#L3820-L3830